ui/src/app/pages/admin/admin-user.tsx

```diff
@@ -132,36 +148,202 @@ const AdminUser = withUrlParams()(class extends React.Component<Props, State> {
-        institution => {
-          return {
-            'label': institution.displayName ? institution.displayName : institution.shortName,
-            'value': {displayName: institution.displayName, shortName: institution.shortName}
-          };
-        },
-        institutionsResponse.institutions
-      );
-      this.setState({verifiedInstitutionOptions: options});
+      const institutionsResponse = await institutionApi().getPublicInstitutionDetails();
+      const institutions = institutionsResponse.institutions;
+      this.setState({verifiedInstitutionOptions: fp.sortBy( 'displayName', institutions)});
     } catch (error) {
       this.setState({institutionsLoadingError: 'Could not get list of verified institutions - please try again later'});
     }
   }
 
+  getRoleOptionsForProfile() {
+    const {updatedProfile: {verifiedInstitutionalAffiliation}, verifiedInstitutionOptions} = this.state;
+    const institutionShortName = verifiedInstitutionalAffiliation ? verifiedInstitutionalAffiliation.institutionShortName : '';
+    return getRoleOptions(verifiedInstitutionOptions, institutionShortName);
+  }
+
+  getInstitutionDropdownOptions() {
+    const {verifiedInstitutionOptions} = this.state;
+    return fp.map(({displayName, shortName}) => ({label: displayName, value: shortName}), verifiedInstitutionOptions);
+  }
+
+  isSaveDisabled(errors) {
+    const {oldProfile, updatedProfile} = this.state;
+    return fp.isEqual(oldProfile, updatedProfile) || errors;
+  }
+
+  renderCheckEmailResponse() {
+    const {checkEmailResponse, updatedProfile, verifiedInstitutionOptions} = this.state;
+    if (updatedProfile && updatedProfile.verifiedInstitutionalAffiliation) {
+      if (checkEmailResponse.isValidMember) {
+        return null;
+      } else {
+        const {verifiedInstitutionalAffiliation} = updatedProfile;
+        const institution = fp.find(
+            institution => institution.shortName === verifiedInstitutionalAffiliation.institutionShortName,
+            verifiedInstitutionOptions
+        );
+        if (institution.duaTypeEnum === DuaType.RESTRICTED) {
+          // Institution has signed Restricted agreement and the email is not in allowed emails list
+          return <RestrictedDuaEmailMismatchErrorMessage/>;
+        } else {
+          // Institution has MASTER or NULL agreement and the domain is not in the allowed list
+          return <MasterDuaEmailMismatchErrorMessage/>;
+        }
+      }
+    }
+    return null;
+  }
+
+  async setVerifiedInstitutionOnProfile(institutionShortName: string) {
+    const {verifiedInstitutionOptions} = this.state;
+    await this.setState({loading: true});
+    await this.setState(fp.flow(
+      fp.set(['updatedProfile', 'verifiedInstitutionalAffiliation', 'institutionShortName'], institutionShortName),
+      fp.set(
+          ['updatedProfile', 'verifiedInstitutionalAffiliation', 'institutionDisplayName'],
+          verifiedInstitutionOptions.find(
```

Comment:
nit: Do we need to grab the property after the find when setting the `institutionDisplayName`

Reply:
eek! yes